Update python_versions.json for python 3.13

ChandraDevs1 opened the pull request:
### **User description**
#7874 

Updated python_version.json file and added latest version of python


___

### **PR Type**
Enhancement, Documentation


___

### **Description**
- Added support for Python 3.13 in `python_versions.json`.

- Updated configuration to include Python 3.13 compatibility.


___



### **Changes walkthrough** 📝
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Configuration changes</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>python_versions.json</strong><dd><code>Added Python 3.13 to supported versions</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

.github/workflows/python_versions.json

<li>Added Python 3.13 to the supported versions list.<br> <li> Ensured compatibility with the latest Python release.


</details>


  </td>
  <td><a href="https://github.com/OpenMS/OpenMS/pull/7877/files#diff-4b1b94e5afdf06ddf41dc657052754a03220657a2749924440fc3d205c140f43">+1/-1</a>&nbsp; &nbsp; &nbsp; </td>

</tr>
</table></td></tr></tr></tbody></table>

___

> <details> <summary>  Need help?</summary><li>Type <code>/help how to ...</code> in the comments thread for any questions about Qodo Merge usage.</li><li>Check out the <a href="https://qodo-merge-docs.qodo.ai/usage-guide/">documentation</a> for more information.</li></details>

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

- **Chores**
  - Expanded runtime support to include Python 3.13, enhancing compatibility and ensuring a smoother development and deployment experience for users.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

ChandraDevs1 committed:
Update python_versions.json for python 3.13

jpfeuffer commented on .github/workflows/python_versions.json:
```diff
@@ -1,3 +1,3 @@
 [
-  "3.9", "3.10", "3.11", "3.12"
+  "3.9", "3.10", "3.11", "3.12","3.13"
```
```suggestion
  "3.9", "3.10", "3.11", "3.12", "3.13"
```

ChandraDevs1 committed:
Update python_versions.json

cbielow reviewed (approved):
nice, thanks!

Now, a complementary PR to pyOpenMS to adapt the documentation (see original issue) would be nice, just to keep things in sync.

ChandraDevs1 commented:
> nice, thanks!
> 
> Now, a complementary PR to pyOpenMS to adapt the documentation (see original issue) would be nice, just to keep things in sync.

Dear @cbielow ,

I have updated the following files to ensure compatibility with Python 3.13.0:
	•	setup.py: Adjusted the python_requires parameter to include Python 3.13.0.     
   link: https://github.com/OpenMS/OpenMS/pull/7875/files
    •	installation.rst: Updated the documentation to reflect support for Python 3.13.0.
        link : https://github.com/OpenMS/pyopenms-docs/pull/455/files
        •       python_version.txt : added the latest version of Python
     link : https://github.com/OpenMS/OpenMS/pull/7877/commits/d191609c231cd36fb504f97fddd28548d4042919

jpfeuffer commented:
I do not agree with:
- putting supported versions in hardcoded docs when this information is easily available in automated ways (https://pypi.org/project/pyopenms/3.3.0/#files)
- artificially restricting the python packaging process to certain python versions, although the build process actually supports more version and it's just the deployment that is lacking in python versions.

poshul commented:
> I do not agree with:
> 
>     * putting supported versions in hardcoded docs when this information is easily available in automated ways (https://pypi.org/project/pyopenms/3.3.0/#files)
> 
>     * artificially restricting the python packaging process to certain python versions, although the build process actually supports more version and it's just the deployment that is lacking in python versions.

I've opened a separate issue to for these improvement suggestions.
